site/_js/firebase.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * @fileoverview Firebase integration.
+ */
+import {initializeApp} from 'firebase/app';
+//import { initializeAppCheck, ReCaptchaV3Provider } from "firebase/app-check";
+import {getDatabase, ref, runTransaction, onValue} from 'firebase/database';
+
+export class Firebase {
+  constructor() {
+    this.config = null;
+    // For local Firebase Emulator
+    if (location.hostname === 'localhost') {
+      this.config = {
+        databaseURL: 'http://localhost:9000/?ns=oct-sip-test2',
+      };
+      // this.config = {
```

Comment:
My understanding is that the only thing we need checked in to the repo is the `apiKey`, which can [safely be shared publicly](https://firebase.google.com/docs/projects/api-keys).

There are a lot of other ids here, all commented out, so I'm not sure which ones you actually plan on checking in. If there's anything other than `apiKey` that needs to be checked in, please clarify. Otherwise, please remove everything that does not need to be checked in, even from the comments.

Reply:
All this will be swapped with production instance configuration settings once it is available.

Reply:
My point is that there shouldn't be any need for either development or production settings for anything other than Cloud Firestore, and we don't plan on using authentication. Many of those options, like `messagingSenderId`, are related to parts of Firebase other than Cloud Firestore, and they should not be checked in, even in a comment.

Reply:
This configuration is required as pert this:
![image](https://user-images.githubusercontent.com/101087073/160234401-92bdf080-67ac-4963-84d4-e6d00c323e6e.png)

Reply:
That configuration is needed if you are using all of those Firebase services, including Messaging, Cloud Storage, etc. We are not using all that for this project. (And if it were required to use the service that we do require via the local emulators, I'm not sure why it would be checked in commented out?)

Reply:
Unless and until we connect this to actual project we need to keep this code commented. Local emulator works without these commented configurations and only requires project login.

We required all the configs except messagingSenderId. If we are going in wrong direction then please provide actual configuration.

Reply:
My point is that we should not be checking in commented out configuration _for services we don't plan on using_.

`storageBucket` is [used to configure Cloud Storage](https://firebase.google.com/docs/storage/web/start). We should not be using Cloud Storage for this project. If there's some reason why we need to have that configuration value checked in and replaced with an actual production value, that makes me concerned that somehow, the code for this project was written assuming that Cloud Storage was being used.

Similarly, we cannot support user authentication for this project. The [`authDomain` setting](https://cloud.google.com/identity-platform/docs/show-custom-domain) is used to configure a custom domain when users go through the authentication flow. If you're saying that, for some reason, we will need to have an `authDomain` value configured in production, that makes me very concerned that the code was written assuming authentication would be required.